tools/compare_benchmark.py

```diff
@@ -0,0 +1,159 @@
+# Runs nvfuser_bench on two commits and compares their results. See
+# https://github.com/NVIDIA/Fuser/wiki/Developer-guide#benchmark-nvfuser for
+# usage.
+
+import argparse
+from collections.abc import Iterable
+from dataclasses import dataclass
+import json
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import subprocess
+
+
+# Runs nvfuser_bench with `benchmark_args` on the given branch or commit. Dumps
+# outputs to `out_dir`. Returns `out_dir`/`branch_or_commit`.json that captures
+# the benchmark result. If the output already exists, skips benchmarking and
+# uses that output. This is useful, for example, when comparing multiple
+# contenders to the same base.
+def run_benchmark(branch_or_commit: str, benchmark_args: str, out_dir: str) -> str:
+    benchmark_out = os.path.join(out_dir, branch_or_commit + ".json")
+    if os.path.exists(benchmark_out):
+        print(f"{benchmark_out} already exists. Skip benchmarking {branch_or_commit}.")
+        return benchmark_out
+
+    subprocess.check_call(f"git checkout {branch_or_commit}", shell=True)
```

Comment:
See here how to silence the detached HEAD warning: https://github.com/NVIDIA/Fuser/blob/main/tools/compare_codegen.sh#L150